analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -486,26 +489,29 @@ class SpdxDocumentFile(
         // For direct callers of this function mapDefinitionFiles() did not populate the map before, so add a fallback.
         val spdxDocument = spdxDocumentForFile.getOrPut(definitionFile) { SpdxModelMapper.read(definitionFile) }
 
-        // Guard against direct callers passing SPDX documents that do not describe a project.
-        val projectPackage = requireNotNull(spdxDocument.projectPackage()) {
-            "The SPDX document file at '$definitionFile' does not describe a project."
-        }
-
-        log.info {
-            "File '$definitionFile' contains SPDX document '${spdxDocument.name}' which describes project " +
-                    "'${projectPackage.name}'."
-        }
-
         val packages = mutableSetOf<Package>()
+        val scopes = sortedSetOf<Scope>()
 
-        val scopes = SPDX_SCOPE_RELATIONSHIPS.mapNotNullTo(sortedSetOf()) { type ->
-            createScope(spdxDocument, projectPackage, type, workingDir, packages)
+        val projectPackage = if (!spdxDocument.isProject()) {
+            spdxDocument.packages[0]
+        } else {
+            requireNotNull(spdxDocument.projectPackage()) {
+                "The SPDX document file at '$definitionFile' does not describe a project."
+            }
         }
+            log.info {
```

Comment:
The formatting is now off (wrong indentation).

IIRC, in the original version, the scopes and packages have only been populated in the `else` branch; now this is always the case. Are you sure that this is correct?

Reply:
I checked again in the example provided by @tsteenbe and that is how it was before. So I think this behaviour is actually desired.

``` json
   {
        "id" : "SpdxDocumentFile::Boost.Geometry:1.70.0",
        "definition_file_path" : "src/include/third_party/boost_geometry/package.spdx.yml",
        "declared_licenses" : [ "BSL-1.0" ],
        "declared_licenses_processed" : {
          "spdx_expression" : "BSL-1.0"
        },
        "vcs" : {
          "type" : "",
          "url" : "",
          "revision" : "",
          "path" : ""
        },
        "vcs_processed" : {
          "type" : "Git",
          "url" : "ssh://git@github.com/heremaps/movetk.git",
          "revision" : "cb150d7a6f5d231e08c75c3dfac1b28cc237f1e9",
          "path" : "src/include/third_party/boost_geometry"
        },
        "homepage_url" : "https://www.boost.org/",
        "scopes" : [ {
          "name" : "default",
          "dependencies" : [ ]
        } ]
      }
```

Reply:
@neubs-bsi  I tested it with MoveTK and one of our internal SDKs - it seems to work as we discussed.